app/models/transaction.rb

```diff
@@ -36,6 +41,9 @@ class Transaction < ActiveRecord::Base
   before_save :check_negative
   after_save :recalculate_amount
   after_destroy :recalculate_amount
+  before_restore do
```

Comment:
@shir If we dont put it here when we restore transaction, bank account balance won't be recalculated and his sum will be like there is no such transaction. Will be outdated